deepchecks/core/serialization/check_result/html.py

```diff
@@ -171,210 +233,194 @@ def prepare_conditions_table(
+            flag indicating that the 'CheckResult' output will be embedded
+            into the 'SuiteResult' output. This flag tells the serializer
+            how to serialize Plotly figures and whether navigation links
+            should be included in the output.
+            This parameter does not have any effect if the 'full_html'
+            parameter was set to 'True'
         is_for_iframe_with_srcdoc : bool, default False
             anchor links, in order to work within iframe require additional prefix
             'about:srcdoc'. This flag tells function whether to add that prefix to
             the anchor links or not
+        use_javascript : bool , default True
+            whether to use  javascript in an output or not.
+            If set to 'False', all components that require javascript
+            to work will be replaced by plain HTML components (if possible).
+            For example, plotly figures will be transformed into JPEG images,
+            tabs widgets will be replaced by HTML '<details>' tag
 
         Returns
         -------
         str
         """
-        return DisplayItemsHandler.handle_display(
-            self.value.display,
-            output_id=output_id,
-            plotly_to_image=plotly_to_image,
-            is_for_iframe_with_srcdoc=is_for_iframe_with_srcdoc
-        )
+        output = [CommonHtml.additional_output_header]
+
+        if len(self.value.display) == 0:
+            output.append(CommonHtml.empty_content_placeholder)
+        else:
+            output.append(''.join(self._display_serializer.serialize(
+                output_id=output_id,
+                embed_into_iframe=embed_into_iframe,
+                embed_into_suite=embed_into_suite,
+                use_javascript=use_javascript,
+                is_for_iframe_with_srcdoc=is_for_iframe_with_srcdoc
+            )))
+
+        if (
+            output_id is not None
+            and embed_into_suite is not None  # whether was embedded into suite output
+        ):
+            output.append(go_to_top_link(
+                output_id=output_id,
+                is_for_iframe_with_srcdoc=is_for_iframe_with_srcdoc
+            ))
 
+        return f'<section data-name="additional-output">{"".join(output)}</section>'
 
-class DisplayItemsHandler(ABCDisplayItemsHandler):
-    """Auxiliary class to decouple display handling logic from other functionality."""
 
-    @classmethod
-    def handle_display(
-        cls,
-        display: t.List['check_types.TDisplayItem'],
+class DisplayItemsSerializer(ABCDisplayItemsSerializer[str]):
+    """CheckResult display items serializer."""
+
+    def serialize(
+        self,
         output_id: t.Optional[str] = None,
         is_for_iframe_with_srcdoc: bool = False,
-        include_header: bool = True,
-        include_trailing_link: bool = True,
+        embed_into_iframe: bool = False,
```

Comment:
why do we need both `embed_into_iframe` and `is_for_iframe_with_srcdoc`?

Reply:
for cases like with `colab`
it inserts output into the iframe by itself so `embed_into_iframe` will not be appropriate here
unfortunately, navigation links do not work in collab even with `about:srcdoc`

Reply:
so maybe if it anyway doesn't work we can just use "embed_into_iframe"?